Feature/comprehensive integration testing

tshepo-OMbank opened the pull request.

tshepo-OMbank committed:
Add integration tests for basic workflow and error handling in simple.test.js and basic-workflow.test.js; enhance global setup and teardown for database management

tshepo-OMbank committed:
Add simple integration tests for API endpoints, error handling, and health check in simple.test.js

tshepo-OMbank committed:
Add integration tests for mock services in mock-services.test.js

tshepo-OMbank committed:
Add integration tests for Commercial Bank mock service, including connection and payment request handling in mock-services.test.js

tshepo-OMbank committed:
Add integration tests for stock management API endpoints, including case stock information, stock reports, and error handling for invalid requests in stock-management.test.js

tshepo-OMbank committed:
Add integration tests for Hand and Bulk Logistics mock services, including health checks and request handling in mock-services.test.js

tshepo-OMbank committed:
Add end-to-end integration tests for order workflow and simulation requests in mock-services.test.js, enhancing coverage of mock services interactions.

tshepo-OMbank committed:
Add health check integration tests for mock services in mock-services.test.js, verifying service status and logging results for debugging.

tshepo-OMbank committed:
Add integration tests for reporting functionality in reporting.test.js, utilizing supertest for API requests and including status code checks.

tshepo-OMbank committed:
Add integration test for bank balance report endpoint in reporting.test.js, verifying response status and structure using supertest.

tshepo-OMbank committed:
Add integration test for stock reports endpoint in reporting.test.js, verifying response status and handling for various scenarios using supertest.

tshepo-OMbank committed:
Add integration test for orders reports endpoint in reporting.test.js, verifying response status and handling for various scenarios using supertest.

tshepo-OMbank committed:
Add integration tests for shipments and transactions reports endpoints in reporting.test.js, verifying response status and handling for various scenarios using supertest.

tshepo-OMbank committed:
Add integration tests for cases, sales, case orders, and case orders stats reports endpoints in reporting.test.js, verifying response status and handling for various scenarios using supertest.

tshepo-OMbank committed:
Add integration tests for simulation date reports and error handling in reporting.test.js, verifying response status for valid and invalid report requests using supertest.

tshepo-OMbank committed:
Add integration test for logistics delivery requests in logistics.test.js, verifying response status and API structure using supertest.

tshepo-OMbank committed:
Add integration test for logistics pickup requests in logistics.test.js, verifying response status and API structure using supertest.

tshepo-OMbank committed:
Add integration test for handling invalid logistics requests in logistics.test.js, verifying response status for missing required fields using supertest.

tshepo-OMbank committed:
Add integration test for handling malformed logistics requests in logistics.test.js, verifying response status for invalid JSON data using supertest.

tshepo-OMbank committed:
Add integration tests for logistics machine delivery and endpoint availability in logistics.test.js, verifying response status and API structure using supertest.

tshepo-OMbank committed:
Remove obsolete integration tests for basic API endpoints and error handling in simple.test.js, streamlining the test suite.

tshepo-OMbank committed:
Remove obsolete health check integration tests from health.test.js, streamlining the test suite.

tshepo-OMbank committed:
Refactor integration tests for clarity and consistency in comments across various test files, enhancing readability and maintaining focus on response handling scenarios.

Liam-Barkley-BBD commented on server/tests/integration/mock-services.test.js:
```diff
@@ -0,0 +1,258 @@
+describe("Mock Services Integration Test", () => {
+  // Mock service endpoints for testing
+  const MOCK_SERVICES = {
+    commercialBank: "http://localhost:3001",
```
Does this run when we deploy?

tshepo-OMbank commented on server/tests/integration/reporting.test.js:
```diff
@@ -0,0 +1,243 @@
+import request from "supertest";
+import { StatusCodes } from "http-status-codes";
+import { app } from "../../server.js";
```
Yup it does, we usually test everything for my client work, but if I could remove it, if it's not relevant

tshepo-OMbank committed:
Merge branch 'dev' of https://github.com/miniconomy2025/Case-Closed-Inc into feature/comprehensive-integration-testing

tshepo-OMbank committed:
Update Jest configuration, add cross-env dependency, and refactor server initialization for test environment compatibility

tshepo-OMbank committed:
Refactor test suite by consolidating validation checks and enhancing error handling to improve robustness and maintainability.

tshepo-OMbank committed:
Enhance Jest integration configuration to support ES modules, including Babel transformation and updated module file extensions for improved test compatibility.

tshepo-OMbank committed:
Update integration test command in package.json to use specific Jest configuration file for improved test execution.

tshepo-OMbank committed:
Add integration test database setup with reset and seed functions

tshepo-OMbank committed:
Add integration tests for database connection, verifying successful connection, database name, migrations table, and required tables.

tshepo-OMbank committed:
Refactor global setup for integration tests to use test database, enhance error handling, and log migration status

tshepo-OMbank committed:
Enhance logistics integration tests with comprehensive scenarios for delivery and pickup requests, including validation for non-existent orders, missing items, and excessive quantities. Implement setup and teardown for test data to ensure consistent test execution.

tshepo-OMbank committed:
Implement comprehensive integration tests for stock management, including setup and teardown for test data, validation of stock retrieval, handling of stock updates after delivery, and scenarios for low and zero stock. Enhance test structure for clarity and maintainability.

tshepo-OMbank committed:
Enhance integration tests for mock services with improved setup and teardown, validation of service availability, and detailed logging. Add tests for payment processing, error handling, and endpoint verification, ensuring robust coverage for various scenarios.

tshepo-OMbank committed:
Enhance orders CRUD integration tests with improved setup and teardown, validation for order creation, retrieval, payment, pickup, and cancellation workflows. Implement detailed logging and error handling for various scenarios, ensuring comprehensive coverage of order lifecycle and stock management.

tshepo-OMbank committed:
Refactor pickup worker to conditionally initialize SQS client based on environment configuration, adding logging for unconfigured states to improve clarity and maintainability.

tshepo-OMbank committed:
Update integration test command in package.json to include database connection parameters for improved test environment configuration.

tshepo-OMbank committed:
Add comprehensive integration tests for simulation and machine failure workflows, including start, stop, and error handling scenarios. Implement setup and teardown for test data, ensuring robust validation of API responses and stock management during simulations.

tshepo-OMbank committed:
Configure Jest integration tests to run sequentially by setting maxWorkers to 1, preventing database conflicts during test execution.

tshepo-OMbank committed:
Refactor test database configuration in integration tests to use environment variables for improved flexibility and security. Load environment variables using dotenv for dynamic database connection settings.

tshepo-OMbank committed:
Refactor stock management integration tests to ensure consistent test setup by cleaning up pending orders and setting known stock values before each test. Update assertions for clarity and accuracy in stock depletion and reserved stock calculations.

tshepo-OMbank committed:
Refactor integration test setup to streamline environment configuration and improve logging. Remove redundant comments and clarify the setup process for test database connections.

tshepo-OMbank committed:
Refactor orders CRUD integration test setup to clean up all pending test orders before each test, ensuring a consistent and clean state for testing. Update comments for clarity on the cleanup process.

tshepo-OMbank committed:
Remove basic workflow integration test file to streamline test suite and eliminate redundancy. This change helps maintain a cleaner codebase by removing outdated or unnecessary tests.

tshepo-OMbank committed:
Remove comprehensive reporting integration test file to streamline the test suite and eliminate redundancy. This change helps maintain a cleaner codebase by removing outdated or unnecessary tests related to report endpoints.

tshepo-OMbank committed:
Add order existence checks in logistics integration tests to ensure consistent test setup. Create new orders if deleted by previous tests, maintaining expected order status for pickup scenarios.

tshepo-OMbank committed:
Enhance global setup for integration tests by adding dotenv import for environment variable management. Improve logging to display the test database name and port, ensuring clearer visibility during test execution.

tshepo-OMbank committed:
Add GitHub Actions workflow for integration, unit tests, and linting

This commit introduces a new GitHub Actions workflow that automates the execution of integration tests, unit tests, and code linting. The integration tests utilize a PostgreSQL service and mock services, ensuring a comprehensive testing environment. The workflow also includes a PR status check that comments on the PR with the results of the tests, enhancing visibility and feedback for developers.
